mmdeploy/apis/core/pipeline_manager.py

```diff
@@ -0,0 +1,357 @@
+        self._call_id += 1
+        return pipe_caller(*args, **kwargs)
+
+    def call_function_async(self, func_name: str, *args, **kwargs) -> int:
+        """call pipeline function.
+
+        Args:
+            func_name (str): function name to be called.
+
+        Returns:
+            int: Call id of this function
+        """
+        pipe_caller = self.get_caller(func_name)
+        assert pipe_caller.is_multiprocess, \
+            f'multiprocess of {func_name} has not been enabled.'
+
+        call_id = self._call_id
+        pipe_caller._call_id = call_id
+        self._call_id += 1
+        proc = mp.Process(target=pipe_caller, args=args, kwargs=kwargs)
+        proc.start()
+        self._proc_async[call_id] = (func_name, proc)
+
+        return call_id
+
+    def get_result_sync(self, call_id: int):
+        """get result of async call."""
+        assert call_id in self._proc_async, f'Unknown call id: {call_id}'
+        func_name, proc = self._proc_async.pop(call_id)
+        proc.join()
+        ret = self.get_caller(func_name).pop_mp_output(call_id)
+
+        return ret
+
+    def call_function(self, func_name: str, *args, **kwargs) -> Any:
+        """call pipeline function.
+
+        Args:
+            func_name (str): function name to be called.
+
+        Returns:
+            Any: The result of call function
+        """
+        pipe_caller = self.get_caller(func_name)
+
+        if self._enable_multiprocess and pipe_caller.is_multiprocess:
+            call_id = self.call_function_async(func_name, *args, **kwargs)
+            if pipe_caller._mp_async:
+                return PipelineResult(self, call_id)
+            return self.get_result_sync(call_id)
+        else:
+            return self.call_function_local(func_name, *args, **kwargs)
+
+    def register_pipeline(self,
+                          func_name: str = None,
+                          is_multiprocess_available: bool = True):
+        """register the pipeline function."""
+
+        def _register(func):
+            assert isinstance(func, Callable), f'{func} is not Callable.'
+            func_name_ = func_name if func_name is not None \
+                else _get_func_name(func)
+
+            # create global function to save the implementation
+            impl_name = f'_pipe_{func.__name__}__impl_'
+            func.__globals__[impl_name] = func
+
+            # create caller
+            pipe_caller = PipelineCaller(
+                func.__module__,
+                impl_name,
+                func_name=func_name_,
+                is_multiprocess_available=is_multiprocess_available)
+            PIPELINE_MANAGER._callers[func_name_] = pipe_caller
+
+            # wrap call
+            @wraps(func)
+            def _wrap(*args, **kwargs):
+                return self.call_function(func_name_, *args, **kwargs)
+
+            return _wrap
+
+        return _register
+
+
+PIPELINE_MANAGER = PipelineManager()
+
+
+class no_mp:
```

Comment:
how about ban_mp or mp_banned?

Reply:
I want to aligned with `no_grad` in PyTorch.